src/command/history.js

```diff
@@ -25,7 +25,7 @@ class HistoryCommand extends Command {
 
     static record(query, result) {
         // Ignore the command history
-        if (!query || query.startsWith(":") || !result) return;
```

Comment:
Let's add a callback argument for `record()` to give the caller the right to decide what kind of result they don't care about. 

Such as:
```js
static record(query, result, ignoreCallback) {
    // Ignore the command history
    if (ignoreCallback && ignoreCallback(result)){
        return;
    }
}
```

What do you think?

Reply:
I think in general that would be sensible, but the callback would also have to include the query to allow a caller to make the same decision that the static-function currently makes, right?

Reply:
How about move the condition-check statements out of the `HistoryCommand`, such as:
```js
        afterNavigated: (query, result) => {
            if (!query || query.startsWith(":") || !result) return;
            HistoryCommand.record(query, result);
        }
```
For clean architecture purposes, this is better than the old one I think.

Reply:
Yes, that sounds sensible; I'll incorporate this!

Reply:
I have removed this check altogether, which means this is is now fully up to the caller!